### ISSUE TITLE
Publish schedule & Speakers

### ISSUE THREAD

**Issue by queryluke** [code4lib/2016.code4lib.org#242]
Work on this has started in [final-schedule-speakers](https://github.com/code4lib/2016.code4lib.org/tree/final-schedule-speakers)

I'll be sort of brain dumping information here:
## Important Files

(they're protected. message me if you need access)
1. [Results from speaker bios](https://drive.google.com/open?id=1KZjkOXYOybheOPC7h5dFLYfNHdiV29ENUK_hhpwOyis) (highlighted rows = all speaker data is collected)
2. [Final Talk list](https://drive.google.com/open?id=1bNGw49eOM1RjmqVjzNbZOcekX_1jnJpK8b3cXrPhF8A)
3. Final Workshop list ( @hackmastera has this. Unless there needs to be another import, it's not currently needed, just mentioning it for documentation purposes)
## Importing
1. As new speakers add their info, I'll map their email address to their talk in 2
2. Then I need to reimport both 1 & 2, push, merge, etc
   - Note: Sadly, yml doesn't allow a leading "@" So when, importing the speaker, have to do a find and replace for the twitter handle. Made a to do below.
## Pictures
1. Speakers that emailed me their picture will need to have the images uploaded to the repo
   - THEN, need to take the relative url and add it to the speaker Sheet (so it gets imported)
2. Speakers whose picture look weird: I'll download, crop and follow the pattern of 1
## Current status

"final" draft ready for launch 2/1/2016
## To do:
1. ~~Code day 2 and 3 into the schedule.yml file~~
2. ~~incorporate talk metadata (group, time, length) into _layout/presentation.html~~
3. ~~Rewrite speaker.html~~
4. ~~Figure out how to link to speakers from schedule/presentation pages~~
5. ~~When we have all the speaker information, have the workshop pages point to speakers_final.yml~~
   - created a new issue for this
6. ~~write script to remove @ from twitter handle and \n or \r from bio to ease import process~~
7. ~~Flex box (on presentation pages)~~
8. ~~flexbox bug on speaker pages for mobile displays (need to set order via javascript)~~
9. ~~link to speaker presentation~~
## Questions:
1. ~~Do we want to publish the schedule as soon as it's ready (probably this weekend)?~~
   - going to go as soon as it's ready. we shooting for this Monday at some time. David will prepare an email and we'll coordinate the sending of the email with the pull
2. ~~Are we going to want descriptions and/or dedicated pages for the breakout sessions / lightning talks?~~
   - for breakouts, no; for lightning, yes. but we can collect that information later
3. ~~Do we want names below the speaker pictures. We'll have names on the speaker page, but on the presentation and schedule pages we have a circle div with a hover animation that shows the name. Should we display the name and remove the animation? or only display the name on small screens ( < 768 which amounts to a tablet, i.e. no mouse.~~
   - yes

**Comment by sdellis** [comment]
I don't think we discussed Question #2 at all yet, but I say no.  This is usually a giant pad of paper or whiteboard in the physical space for sign ups.  No need to complicate it with adding web pages.

**Comment by queryluke** [comment]
On this branch, schedule.html will be replaced with /schedule. It's essentially the same content with the following changes:
1. Past keynotes have been removed
2. links have been added to the individual days

**Comment by queryluke** [comment]
I've updated the top navigation. In the navbar, schedule is now a dropdown with Day 1, Day 2, Day 3, Workshops, Overview.

Footer "schedule" link, goes to "Overview", i.e. the new schedule page.

**Comment by queryluke** [comment]
I updated the speaker page. Any design oriented people should feel free to take liberties with it.

**Comment by sdellis** [comment]
Great first draft @roastduckalamode .  Thank you!  

A few things that I will try to get to this weekend (but feel free to beat me to it!): 
1. Overview and Workshop should probably be listed above the Days in the drop down
2. I don't like that you have to hover over the speaker image to get the name... to answer your question, yes remove the animation and display speaker name next to image.
3. I want to see some zebra striping to separate talks on the timelines
4. Speaker detail info on right (Date, Group (?), Time, Length) seems floating. Box it up.
5. You said you updated the speaker page (speakers.html?) but I don't see any change... did I miss something?
6. Are we still waiting on speaker images? Can we either use twitter pics or have a default if we have none for consistency when we announce? 

If you want to do a Google Hangout to go over any questions, let me know.  I'm going to be working on this stuff (and the reception page) from afternoon to night this Sunday (packed sched for me on Sat and Sun mornings).

**Comment by queryluke** [comment]
@sdellis Silly, silly me. I forgot to commit the changes. They're up there now. I'll get the changes you mentioned done today so you can focus on minor edits tomorrow. 

RE: 4 (and purely for my enlightenment) I'm using the default bootstrap col- (which use float). But you've mentioned before to use flex-box. I'm no css expert, so what's the benefit?

I'll be available tonight or tomorrow for a hangout if need be.

**Comment by queryluke** [comment]
@sdellis From your suggestions, 1,2,3,6 are done. And I've added more speakers.

Note about 6: If the talk doesn't have any speaker matches, it renders whatever text is in the speakers column of the google sheet. This reflects the original value enter from the talk proposal. You'll notice there is no anchor link (cause there is no one to link to). I spoke with David and he feels this will suffice for the time being. At least we'll have the speaker names.

RE 4, again: I'll have to learn more about flexbox. I saw what you did for the workshops and talks using .sequence. But I'll need to learn more. My next task will be figuring out a way to show the additional speaker info (right now via modal) on the speakers page. The modal works fine, but when you click on a speaker from the schedule page (or talk page), it would be nice to have the modal open automatically. Or get rid of the modal and use some homebaked solution. That's what I'll be working on next, so feel free to start with the flexbox issues.

**Comment by sdellis** [comment]
Awesome. I'm at a conference all day today, but I will def. look at the flexbox issues after my 8 year old's birthday party tomorrow!

**Comment by queryluke** [comment]
:birthday:

**Comment by queryluke** [comment]
besides the flexbox on the presentation pages, we should be gtg. I'll continue to re-import the speakers every day.

**Comment by sdellis** [comment]
Thanks @roastduckalamode I was just getting to this... (been a long weekend). Looking now...

**Comment by queryluke** [comment]
@sdellis fyi, I did learn how to use flexbox (see the speakers page, it's all flexy). But I didn't get around to the presentation pages. I can get to those this week, but not before the launch tomorrow.

**Comment by sdellis** [comment]
@roastduckalamode ok, take a look at my most recent commit on this branch.  I cleaned up the design.  I really like what you did with the speakers page (love the 404 images!).  I didn't think flexbox was actually necessary in the presentation pages.  I just "boxed it up" to make it pop.  I just realized the workshops pages need the same attention... will get to that after dinner. ;)

However, one thing I DO NOT LIKE is this grouping thing.  It's misinformation, and we need to put the exact times each talk is going on, not when the group starts.  Is there a way we can do this?  I also do not like having "2016" in the date. Everyone knows what year it is and it wraps awkwardly.  Can you help there as well?

**Comment by queryluke** [comment]
@sdellis sure thing. The groupings came from [this document](https://drive.google.com/open?id=1Gl-tunWoV1scwztejM83nynyPpfBJEYaX4Xz_8vhstE) that was sent out to the speakers. I was operating under the impression that it was the final document.

Are you saying you want the specific time on the individual talk pages? or on the schedule page, beside each talk (within a group), you want the time. The former won't be too difficult. The later might take a bit of time.

And I'll look into removing "2016" in a bit.

**Comment by queryluke** [comment]
@sdellis just pulled down your commit. Love the "pop" :+1: 

I removed the "2016". And mirrored the workshop info as you had it for the talks, but the location wrapping oddly. Don't know if you want an additional section or to delineate room in a different way.

I also realized, I think you meant on the individual talk pages, not the schedule page. So I'm going to grab some dinner myself then start on that.

Luke

**Comment by queryluke** [comment]
@sdellis Okiedok. I've got timestamps for all the talks loaded. I used timestamps because we can reformat the time however with simple changes to the code. For the presentation pages, I just put the start time. Seemed to make sense with the length right below. But if you want to mess around with what shows up in that section, the start time is `page.startTime` and end is `page.endTime`

You can reformat using `{{ page.startTime | date: "%values" }}`

replace **%values** with any of the following, and you can toss punctuation in between the values ([below is copied from this](http://cheat.markdunkley.com/)):

%a - The abbreviated weekday name (`Sun'')
%A - The full weekday name (`Sunday'')
%b - The abbreviated month name (`Jan'')
%B - The full month name (`January'')
%c - The preferred local date and time representation
%d - Day of the month (01..31)
%H - Hour of the day,24-hour clock (00..23)
%I - Hour of the day,12-hour clock (01..12)
%j - Day of the year (001..366)
%m - Month of the year (01..12)
%M - Minute of the hour (00..59)
%p - Meridian indicator (`AM'' or`PM'')
%S - Second of the minute (00..60)
%U - Week number of the current year, starting with the first Sunday as the first day of the first week (00..53)
%W - Week number of the current year, starting with the first Monday as the first day of the first week (00..53)
%w - Day of the week (Sunday is 0,0..6)
%x - Preferred representation for the date alone,no time
%X - Preferred representation for the time alone,no date
%y - Year without a century (00..99)
%Y - Year with century
%Z - Time zone name
%% - Literal ``%'' character

**Comment by queryluke** [comment]
Oh and the speakers who had their emails listed opted for that on the form. Should we put it back in? It can always be done later. Food for thought. 

FYI - I am done for the night. Early day tomorrow. But leave whatever comments or fixes and I'll get to them in the morning. Hopefully we can publish by the afternoon.

**Comment by sdellis** [comment]
Awesome!  It all looks great and thanks for making those changes.  We also need to get the keynote speakers and reception info in.  I will try to get that done tonight. I made a minor tweak to the footer that I will bundle with it.  

Other than that, what else needs to happen before publishing?  Are you still waiting on stuff from Dave/Programming Comittee?

**Comment by queryluke** [comment]
I'm not waiting on anything from Dave. When we talked on Thursday, all he wanted was a schedule that reflected the one in the comment above w/ speaker names. So I think we've gone above and beyond. 

Sorry about the keynotes. I didn't have any of their info. I think @bibliotechy said he would use the speaker form to "upload" their info. If you want to hard code it tonight, I'll program it in later. We'll need to do the same with the lightning talks once the program committee decides how many/who. I guess I was waiting on that...but it wasn't essential to publish for tomorrow.

**Comment by queryluke** [comment]
@sdellis @bibliotechy I've added the keynote section. I used a 404 image for kate and grabbed a snippet for her bio from the code4lib wiki. That should suffice until we get her info?

Anything else need to be done?

**Comment by sdellis** [comment]
@roastduckalamode I don't like the idea of making an announcement that includes the Keynotes without getting their up-to-date info.  I propose we hold off on the announcement until we hear back from Kate (but let her know we want to move).  There's no reason to rush this to the expense of having a "404" for a keynote.

**Comment by bibliotechy** [comment]
I let her know we needed it, but haven't heard anything yet.
